hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/BaseLoadBalancer.java

```diff
@@ -1639,6 +1641,44 @@ private void roundRobinAssignment(Cluster cluster, List<RegionInfo> regions,
     }
   }
 
+  private Map<ServerName, List<RegionInfo>> toEnsumbleTableClusterLoad(
+      Map<TableName, Map<ServerName, List<RegionInfo>>> clusterLoadPerTable) {
+    Map<ServerName, List<RegionInfo>> returnMap = new TreeMap<>();
+    for (Map<ServerName, List<RegionInfo>> serverNameListMap : clusterLoadPerTable.values()) {
+      serverNameListMap.forEach((serverName, regionInfoList) -> {
+        List<RegionInfo> regionInfos =
+            returnMap.computeIfAbsent(serverName, k -> new ArrayList<>());
+        regionInfos.addAll(regionInfoList);
+      });
+    }
+    return returnMap;
+  }
+
+  @Override
+  public List<RegionPlan> balanceTable(TableName tableName, Map<ServerName, List<RegionInfo>> clusterLoad){
+    throw new UnsupportedOperationException("balanceTable is not support in BaseLoadBalancer");
+  }
+
+  @Override
+  public synchronized List<RegionPlan>
```

Comment:
Why synchronized?

Reply:
method of the extends class need be synchronized